### PR TITLE
OH2-189 Add page info to paginated endpoints

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-189.

### CODE REVIEW

File: src/main/java/org/isf/examination/service/ExaminationOperations.java [hunk 1]
```diff
@@ -89,10 +92,15 @@ public PatientExamination getLastByPatID(int patID) throws OHServiceException	{
 
 	public List<PatientExamination> getLastNByPatID(int patID, int number) throws OHServiceException {
 		if (number > 0) {
-			return repository.findByPatient_CodeOrderByPexDateDesc(patID, PageRequest.of(0, number)).getContent();
+			return repository.findByPatient_CodeOrderByPexDateDesc(patID, PageRequest.of(0, number));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/service/ExaminationOperations.java:95]: With all these changes to methods it really would be good to add unit tests that verify they are doing what is expected, IMHO.

Comment (ArnaudFonzam) [reply src/main/java/org/isf/examination/service/ExaminationOperations.java:95]: okay your suggestion is to add test method for all methods that I added.... right?

Comment (ArnaudFonzam) [reply src/main/java/org/isf/examination/service/ExaminationOperations.java:95]: the difference at the the new method return paginated data with the page information. early we just return data without page info.

File: src/test/java/org/isf/admission/test/Tests.java [hunk 2]
```diff
@@ -272,6 +274,44 @@ public void testIoGetAdmittedPatientsShouldFindByOneOfFieldsLikeFirstName() thro
...
+	
+	@Test
+	public void testIoGetAdmittionsByDatePageable() throws Exception {
+		// given:
+		int id = setupTestAdmission(false);
+		String str = "2000-01-01T10:11:30";
+		String str2 = "2023-05-05T10:11:30";
+		LocalDateTime dateFrom = LocalDateTime.parse(str);
+		LocalDateTime dateTo =  LocalDateTime.parse(str2);
+		System.out.println(dateFrom+" "+dateTo);
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/admission/test/Tests.java:286]: Not a good practice to leave `println` in the code.

File: src/test/java/org/isf/patient/test/Tests.java [hunk 3]
```diff
@@ -103,7 +104,14 @@ public void testIoGetPatientsPageable() throws Exception {
 		List<Patient> patients = patientIoOperation.getPatients(createPageRequest());
 		testPatient.check(patients.get(patients.size() - 1));
 	}
-
+	
+	@Test 
+	public void testIoGetPatientsPageable2() throws Exception {
+	  setupTestPatient(false); PagedResponse<Patient> patients =
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/patient/test/Tests.java:110]: strange formatting; two statements on the same line

File: src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java [hunk 4]
```diff
@@ -38,5 +38,8 @@ public interface ExaminationIoOperationRepository extends JpaRepository<PatientE
 	List<PatientExamination> findByPatient_CodeOrderByPexDateDesc(@Param("patientCode") int patientCode);
 
 	@Query(value = "select p from PatientExamination p where p.patient.code = :patientCode order by p.pex_date desc")
-	Page<PatientExamination> findByPatient_CodeOrderByPexDateDesc(@Param("patientCode") int patientCode, Pageable pageable);
+	List<PatientExamination> findByPatient_CodeOrderByPexDateDesc(@Param("patientCode") int patientCode, Pageable pageable);
+	
+	@Query(value = "select p from PatientExamination p where p.patient.code = :patientCode order by p.pex_date desc")
+	Page<PatientExamination> findByPatient_CodeOrderByPexDateDescs(@Param("patientCode") int patientCode, Pageable pageable);
```
Comment (mwithi) [inline src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java:44]: I don't understand this change, is not exactly the same method only adding one "s" at the end?

Comment (mwithi) [reply src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java:44]: Sorry, my bad, I misreaded the returning type.

Comment (mwithi) [reply src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java:44]: BTW is the "s" really needed?

**Review by mwithi — CHANGES_REQUESTED** [review]
Please avoid duplication of methods, and prefer meaningful names instead of similar ones (like adding "s")

File: src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java (hunk 4 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/examination/service/ExaminationIoOperationRepository.java:44]: okay

File: src/test/java/org/isf/patient/test/Tests.java [hunk 5]
```diff
@@ -103,7 +104,14 @@ public void testIoGetPatientsPageable() throws Exception {
 		List<Patient> patients = patientIoOperation.getPatients(createPageRequest());
 		testPatient.check(patients.get(patients.size() - 1));
 	}
-
+	
+	@Test 
+	public void testIoGetPatientsPageable2() throws Exception {
+	  setupTestPatient(false); 
+	  PagedResponse<Patient> patients = patientIoOperation.getPatientsPageable(createPageRequest());
+	  testPatient.check(patients.getData().get(patients.getData().size() - 1));
```
[the remark this answers is not in the document]
Comment (ArnaudFonzam) [reply src/test/java/org/isf/patient/test/Tests.java:112]: okay

**Review by mwithi — APPROVED** [review]

File: src/test/java/org/isf/patient/test/Tests.java [hunk 6]
```diff
@@ -96,14 +97,14 @@ public void testIoGetPatients() throws Exception {
...
 	}
-
-	@Test
+	
+	@Test 
 	public void testIoGetPatientsPageable() throws Exception {
-		setupTestPatient(false);
-		List<Patient> patients = patientIoOperation.getPatients(createPageRequest());
-		testPatient.check(patients.get(patients.size() - 1));
+	  setupTestPatient(false);
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/patient/test/Tests.java:103]: the previous formatting looks more correct.   Why is there a trailing blank added on line 103 and 105?

File: src/main/java/org/isf/lab/service/LabIoOperations.java [hunk 7]
```diff
@@ -89,16 +91,26 @@ public List<LaboratoryRow> getLabRow(Integer code) throws OHServiceException {
...
-			return pagedResult;
+			return repository.findByLabDateBetweenOrderByLabDateDesc(time1, time2, pageable).getContent();
+
+		}
+		return repository.findAll(pageable).getContent();
+	}
+	
+	public PagedResponse<Laboratory> getLaboratoryPageable(boolean onWeek, int pageNo, int pageSize) throws OHServiceException {
+		Pageable pageable = PageRequest.of(pageNo, pageSize);
+		if (onWeek) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/service/LabIoOperations.java:106]: shouldn't all the occurrences of `onWeek` really be `oneWeek`?

Comment (mwithi) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:106]: I suppose yes, but some days ago in another PR you and @ArnaudFonzam discussed to "decide for one" and that one has been chosen...

**Comment by mwithi** [comment]
@ArnaudFonzam please read the reviews.
A part small things, I ran the tests and all are **ok**. 

BTW, is it correct that none of the new methods are used yet in API? Or at least I don't see any PR paired with this one, correct?

File: src/main/java/org/isf/lab/service/LabIoOperations.java (hunk 7 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:106]: I have used onWeek on the api that is why I also use it on the core.

Comment (ArnaudFonzam) [reply src/main/java/org/isf/lab/service/LabIoOperations.java:106]: yes is it you right okay let me update it on core and on the api

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**